hack/generated/pkg/armclient/template_client.go

```diff
@@ -0,0 +1,322 @@
+		Value TemplateResourceObjectOutput `json:"value,omitempty"`
+	}
+
+	ClientConfig struct {
+		Env    Enver
+		Logger logr.Logger
+	}
+
+	AzureTemplateClientOption func(config *ClientConfig) *ClientConfig
+)
+
+var _ Applier = &AzureTemplateClient{}
+
+func WithEnv(env Enver) func(*ClientConfig) *ClientConfig {
+	return func(cfg *ClientConfig) *ClientConfig {
+		cfg.Env = env
+		return cfg
+	}
+}
+
+func WithLogger(logger logr.Logger) func(*ClientConfig) *ClientConfig {
+	return func(cfg *ClientConfig) *ClientConfig {
+		cfg.Logger = logger
+		return cfg
+	}
+}
+
+func NewAzureTemplateClient(opts ...AzureTemplateClientOption) (*AzureTemplateClient, error) {
+	cfg := &ClientConfig{
+		Env:    new(stdEnv),
+		Logger: ctrl.Log.WithName("azure_template_client"),
+	}
+
+	for _, opt := range opts {
+		opt(cfg)
+	}
+
+	subID := cfg.Env.GetEnv(auth.SubscriptionID)
+	if subID == "" {
+		return nil, errors.Errorf("env var %q was not set", auth.SubscriptionID)
+	}
+
+	envSettings, err := GetSettingsFromEnvironment(cfg.Env)
+	if err != nil {
+		return nil, err
+	}
+
+	authorizer, err := envSettings.GetAuthorizer()
+	if err != nil {
+		return nil, err
+	}
+
+	rawClient, err := NewClient(authorizer)
+	if err != nil {
+		return nil, err
+	}
+
+	return &AzureTemplateClient{
+		RawClient:      rawClient,
+		Logger:         cfg.Logger,
+		SubscriptionID: subID,
+	}, nil
+}
+
+func (atc *AzureTemplateClient) GetResource(ctx context.Context, id string, apiVersion string, status genruntime.ArmResourceStatus) error {
+	if id == "" {
+		return errors.Errorf("resource ID cannot be empty")
+	}
+
+	path := fmt.Sprintf("%s?api-version=%s", id, apiVersion)
+	err := atc.RawClient.GetResource(ctx, path, &status) // TODO: is this right?
+	return err
+}
+
+// CreateDeployment deploys a resource to Azure via a deployment template
+func (atc *AzureTemplateClient) CreateDeployment(ctx context.Context, deployment *Deployment) (*Deployment, error) {
+	return atc.RawClient.PutDeployment(ctx, deployment)
+}
+
+// DeleteDeployment deletes a deployment. If the deployment doesn't exist it does not return an error
+func (atc *AzureTemplateClient) DeleteDeployment(ctx context.Context, deploymentId string) error {
+	err := atc.RawClient.DeleteResource(ctx, idWithAPIVersion(deploymentId), nil)
+
+	// NotFound is a success
+	if IsNotFound(err) {
+		return nil
+	}
+
+	return err
+}
+
+func (atc *AzureTemplateClient) GetDeployment(ctx context.Context, deploymentId string) (*Deployment, error) {
+	var deployment Deployment
+	if err := atc.RawClient.GetResource(ctx, idWithAPIVersion(deploymentId), &deployment); err != nil {
+		return &deployment, err
+	}
+	return &deployment, nil
+}
+
+func (atc *AzureTemplateClient) NewDeployment(resourceGroup string, deploymentName string, resourceSpec genruntime.ArmResourceSpec) *Deployment {
+	//if res.ResourceGroup() == "" {
+	//	return NewSubscriptionDeployment(atc.SubscriptionID, res.Location, name, res)
+	//}
```

Comment:
Should this code be restored or deleted?

Reply:
Replaced with TODO comment